tests/schema/tests.py

```diff
@@ -1769,9 +1765,58 @@ def test_alter_field_add_index_to_textfield(self):
         with connection.schema_editor() as editor:
             editor.alter_field(Note, new_field, old_field, strict=True)
         # Ensure the info constraints where dropped
-        constraints = self.get_constraints(Note._meta.db_table)
-        info_indexes = []
-        for details in constraints.values():
-            if details['columns'] == ['info']:
-                info_indexes.append(details)
+        info_indexes = self.get_constraints_for_column(Note, 'info')
         self.assertEqual(0, len(info_indexes), 'Indexes were not dropped for the info column')
+
+    @unittest.skipUnless(connection.vendor == 'postgresql', "PostgreSQL specific")
+    def test_alter_field_add_unique_to_charfield_with_db_index(self):
+        # Create the table and verify initial number of indexes.
+        with connection.schema_editor() as editor:
+            editor.create_model(BookWithoutAuthor)
+        indexes = self.get_constraints_for_column(BookWithoutAuthor, 'title')
+        self.assertEqual(2, len(indexes))
+
+        # Alter to add unique=True (should add 1 index)
+        old_field = BookWithoutAuthor._meta.get_field('title')
+        new_field = CharField(max_length=100, db_index=True, unique=True)
+        new_field.set_attributes_from_name('title')
+        with connection.schema_editor() as editor:
+            editor.alter_field(BookWithoutAuthor, old_field, new_field, strict=True)
+        indexes = self.get_constraints_for_column(BookWithoutAuthor, 'title')
+        # TODO: check if 3 indexes is correct
+        self.assertEqual(3, len(indexes))
```

Comment:
The fact a normal index and a unique one is tracked by [#24082](https://code.djangoproject.com/ticket/24082). The third one is the `_like` index.